src/%ZPM/PackageManager/Client/REST/PublishClient.cls

```diff
@@ -3,36 +3,66 @@ Class %ZPM.PackageManager.Client.REST.PublishClient Extends (%ZPM.PackageManager
 
 Method PublishModule(pModule As %ZPM.PackageManager.Core.Module) As %Boolean
 {
-  Quit ..Publish(pModule)
+    Quit ..Publish(pModule)
 }
 
 Method PublishApplication(pApplication As %ZPM.PackageManager.Core.Application) As %Boolean
 {
-  Quit ..Publish(pApplication)
+    Quit ..Publish(pApplication)
 }
 
 Method Publish(pModule As %ZPM.PackageManager.Core.VersionedBase) As %Boolean
 {
-  if ((..Username="") || (..Password="")) && (..Token="") {
-    $$$ThrowStatus($$$ERROR($$$GeneralError, "Publishing module, authorization required."))
-  }
-  
-  Set tRequest = ..GetHttpRequest()
-  Set tRequest.ContentType = "application/json"
-  Set publishTimeout =  ##class(%ZPM.PackageManager.Client.Settings).GetValue("publish_timeout")
-  Set:publishTimeout'="" tRequest.Timeout = publishTimeout
-  
-  $$$ThrowOnError(pModule.%JSONExportToStream(tRequest.EntityBody))
-  $$$ThrowOnError(tRequest.Post(tRequest.Location_"packages"))
-  
-  If (tRequest.HttpResponse.StatusCode '= 200) {
-    If tRequest.HttpResponse.StatusCode = 401 {
-      $$$ThrowStatus($$$ERROR($$$GeneralError, "Publishing module, authorization required."))
-    } Else {
-      $$$ThrowStatus($$$ERROR($$$GeneralError, "Publishing module, something went wrong."))
+    If ((..Username="") || (..Password="")) && (..Token="") {
+        $$$ThrowStatus($$$ERROR($$$GeneralError, "Publishing module, authorization required."))
     }
-  }
-  Quit 1
+
+    Set tRequest = ..GetHttpRequest()
+    Set tRequest.ContentType = "application/json"
+    Set publishTimeout =  ##class(%ZPM.PackageManager.Client.Settings).GetValue("publish_timeout")
+    Set:publishTimeout'="" tRequest.Timeout = publishTimeout
+
+    $$$ThrowOnError(pModule.%JSONExportToStream(tRequest.EntityBody))
+    $$$ThrowOnError(tRequest.Post(tRequest.Location_"packages"))
+
+    If (tRequest.HttpResponse.StatusCode '= 200) {
+        If tRequest.HttpResponse.StatusCode = 401 {
+            $$$ThrowStatus($$$ERROR($$$GeneralError, "Publishing module, authorization required."))
+        } Else {
+            $$$ThrowStatus($$$ERROR($$$GeneralError, "Publishing module, something went wrong."))
+        }
+    }
+    Return 1
+}
+
+Method CheckUnpublishEnabled(packageName As %String, packageVersion As %String) As %Boolean
+{
+    If ((..Username="") || (..Password="")) && (..Token="") {
+        $$$ThrowStatus($$$ERROR($$$GeneralError, "Unpublishing module, authorization required."))
+    }
+    Set tRequest = ..GetHttpRequest()
+    $$$ThrowOnError(tRequest.Send("OPTIONS",tRequest.Location_"packages/"_packageName_"/"_packageVersion))
+    If (tRequest.HttpResponse.StatusCode '= 200) {
```

Comment:
I think it's unnecessary here, if any error happens, just silently expect that unpublish Is disabled

Reply:
agree, remove